frontend/src/v5/ui/components/shared/modals/templates/deleteModal/deleteModal.component.tsx

```diff
@@ -14,48 +14,113 @@
+	confidenceCheck?: boolean,
 }
 
-export const DeleteModal = ({ onClickConfirm, onClickClose, title, message }: IDeleteModal) => (
-	<ModalContainer>
-		<CircledIcon variant="error" size="large">
-			<DeleteIcon />
-		</CircledIcon>
-		<DialogTitle>
-			<FormattedMessage
-				id="deleteModal.header"
-				defaultMessage={title}
-			/>
-		</DialogTitle>
-		<DialogContent>
-			<DialogContentText>
-				{message}
-			</DialogContentText>
-		</DialogContent>
-		<Actions>
-			<Button autoFocus type="submit" onClick={() => { onClickClose(); onClickConfirm(); }} variant="contained" color="primary">
-				<FormattedMessage
-					id="deleteModal.action.confirm"
-					defaultMessage="Delete"
-				/>
-			</Button>
-			<Button onClick={onClickClose} variant="outlined" color="secondary">
-				<FormattedMessage
-					id="deleteModal.action.cancel"
-					defaultMessage="Cancel"
-				/>
-			</Button>
-		</Actions>
-	</ModalContainer>
-);
+export const DeleteModal = ({ onClickConfirm, onClickClose, name, message, confidenceCheck }: IDeleteModal) => {
+	const { control, watch, handleSubmit } = useForm({
+		mode: 'onChange',
+		defaultValues: { retypedName: '' },
+	});
+	const isValid = confidenceCheck ? (watch('retypedName') === name) : true;
+	const [errorMessage, setErrorMessage] = useState('');
+	const onError = (error) => {
+		if (isNetworkError(error)) {
+			setErrorMessage(
+				formatMessage({
+					id: 'editProfile.networkError',
+					defaultMessage: 'Network Error',
+				}),
+			);
+		} else {
+			setErrorMessage(error.response.data.message);
+		}
+	};
+
+	const onSuccess = () => {
+		onClickClose();
+	};
+
+	return (
+		<DialogContainer>
+			<CircledIcon variant="error" size="large">
+				<DeleteIcon />
+			</CircledIcon>
+			<DialogTitle>
+				<TruncatableTitle>
+					<FormattedMessage
+						id="deleteModal.header"
+						defaultMessage="Delete {name}?"
+						values={{ name }}
+					/>
+				</TruncatableTitle>
+			</DialogTitle>
+			<Message>
+				<DialogContentText>
+					{message}
+				</DialogContentText>
+				{ confidenceCheck && (
+					<RetypeCheck>
+						<Instruction>
+							<FormattedMessage
+								id="deleteModal.content.retypeCheck"
+								defaultMessage="Confirm by typing {confirmationPhrase} below:"
+								values={{
+									confirmationPhrase: <ConfirmationPhrase>{name}</ConfirmationPhrase>,
+								}}
+							/>
+						</Instruction>
+						<RetypeCheckField
+							control={control}
+							name="retypedName"
+						/>
+					</RetypeCheck>
+				)}
+				{errorMessage && (
+					<ErrorMessage>
+						{errorMessage}
+					</ErrorMessage>
+				)}
+			</Message>
+			<Actions>
+				<Button onClick={onClickClose} variant="contained" color="primary">
+					<FormattedMessage
+						id="deleteModal.action.cancel"
+						defaultMessage="Cancel"
+					/>
+				</Button>
+				<Button autoFocus type="submit" onClick={() => handleSubmit(onClickConfirm(onSuccess, onError))} variant="outlined" color="secondary" disabled={!isValid}>
```

Comment:
if we change onClickConfirm to an async function
```suggestion
				<Button autoFocus type="submit" onClick={handleSubmit(onSubmit))} variant="outlined" color="secondary" disabled={!isValid}>
```

and then somewhere there should be something like 
```
	const onSubmit = async () => {
		try {
			await onClickConfirm();
			onClickClose();
		} catch (e) {
			onError(e);
		}
	};
```